### ISSUE TITLE
new switch-to-configuration-ng: Failed to run user activation... 

### ISSUE THREAD

**Issue by sedlund** [NixOS/nixpkgs#319529]
https://github.com/NixOS/nixpkgs/blame/d2679dcbf1a032b0583915a8e3f9faffe936e9f1/pkgs/by-name/sw/switch-to-configuration-ng/src/main.rs#L1671

when attempting to nixos-rebuild on a wsl instance I am getting this

```log
/nix/store/xj3icmv6yp869q7x0bvkvm5g3jda9b1r-nixos-system-wsl-24.11.20240612.57d6973/bin/switch-to-configuration switch

activating the configuration...
--- diff to current-system
<<< /run/current-system
>>> /nix/store/xj3icmv6yp869q7x0bvkvm5g3jda9b1r-nixos-system-wsl-24.11.20240612.57d6973
No version or selection state changes.
Closure size: 645 -> 645 (0 paths added, 0 paths removed, delta +0, disk usage +0B).
---
setting up /etc...
setting up /bin...
setting up /bin/login...
sops-install-secrets: Imported /etc/ssh/ssh_host_rsa_key as GPG key with fingerprint 6cc739adb08686e46c9a438a7745b1977c437fc1
sops-install-secrets: Imported /etc/ssh/ssh_host_ed25519_key as age key with fingerprint age1w42mwytnct7j978vz9wxn65g74f89uwzx27y8dhjxzkukjnrpdys9df55x
setting up /sbin/init shim...
reloading user units for sedlund...
Error: Failed to run user activation for sedlund

Caused by:
    Permission denied (os error 13)
```

This error message is not great.  How can I find out what is breaking?

tag @jmbaur

**Comment by jmbaur** [comment]
Yeah that error message is not great! What sort of WSL setup do you have? Are you using https://github.com/nix-community/NixOS-WSL?

**Comment by jmbaur** [comment]
I'm not immediately seeing where the program would have failed in such a way that it failed to produce some meaningful error message. With the username change, did your UID happen to change as well? Perhaps it could have failed [here](https://github.com/NixOS/nixpkgs/blob/0bbe0b8728ff5dce6debdd83726d0dd53d4a2d7a/pkgs/by-name/sw/switch-to-configuration-ng/src/main.rs#L1666).

**Comment by sedlund** [comment]
yeah remained 1000:100

maybe this helps. but I don't see anything glaring here either.

<details>
<summary>strace output</summary>
[pid 613199] execve("/nix/store/w5qgrjabiwh5ijm5vmc1kcpn04jfj3nw-switch-to-configuration-0.1.0/bin/switch-to-configuration", ["/nix/store/w5qgrjabiwh5ijm5vmc1kcpn04jfj3nw-switch-to-configuration-0.1.0/bin/switch-to-configuration"], ["DISPLAY=:0", "DISTRO_ID=nixos", "GTK_A11Y=none", "GTK_PATH=/root/.nix-profile/lib/gtk-2.0:/root/.nix-profile/lib/gtk-3.0:/root/.nix-profile/lib/gtk-4.0:/nix/profile/lib/gtk-2.0:/nix/profile/lib/gtk-3.0:/nix/profile/lib/gtk-4.0:/root/.local/state/nix/profile/lib/gtk-2.0:/root/.local/state/nix/profile/lib/gtk-3.0:/root/.local/state/nix/profile/lib/gtk-4.0:/etc/profiles/per-user/root/lib/gtk-2.0:/etc/profiles/per-user/root/lib/gtk-3.0:/etc/profiles/per-user/root/lib/gtk-4.0:/nix/var/nix/profiles/default/lib/gtk-2.0:/nix/var/nix/profiles/default/lib/gtk-3.0:/nix/var/nix/profiles/default/lib/gtk-4.0:/run/current-system/sw/lib/gtk-2.0:/run/current-system/sw/lib/gtk-3.0:/run/current-system/sw/lib/gtk-4.0", "HOME=/root", "INFOPATH=/root/.nix-profile/info:/root/.nix-profile/share/info:/nix/profile/info:/nix/profile/share/info:/root/.local/state/nix/profile/info:/root/.local/state/nix/profile/share/info:/etc/profiles/per-user/root/info:/etc/profiles/per-user/root/share/info:/nix/var/nix/profiles/default/info:/nix/var/nix/profiles/default/share/info:/run/current-system/sw/info:/run/current-system/sw/share/info", "INSTALL_BOOTLOADER=/nix/store/php4qidg2bxzmm79vpri025bqi0fa889-coreutils-9.5/bin/true", "LANG=en_US.UTF-8", "LC_TIME=en_GB.UTF-8", "LIBEXEC_PATH=/root/.nix-profile/libexec:/nix/profile/libexec:/root/.local/state/nix/profile/libexec:/etc/profiles/per-user/root/libexec:/nix/var/nix/profiles/default/libexec:/run/current-system/sw/libexec", "LOCALE_ARCHIVE=/nix/store/fsyf478vqk0d2p0l3g3i6vga3ka0573c-glibc-locales-2.39-52/lib/locale/locale-archive", "LOGNAME=root", "LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=00:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.7z=01;31:*.ace=01;31:*.alz=01;31:*.apk=01;31:*.arc=01;31:*.arj=01;31:*.bz=01;31:*.bz2=01;31:*.cab=01;31:*.cpio=01;31:*.crate=01;31:*.deb=01;31:*.drpm=01;31:*.dwm=01;31:*.dz=01;31:*.ear=01;31:*.egg=01;31:*.esd=01;31:*.gz=01;31:*.jar=01;31:*.lha=01;31:*.lrz=01;31:*.lz=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.lzo=01;31:*.pyz=01;31:*.rar=01;31:*.rpm=01;31:*.rz=01;31:*.sar=01;31:*.swm=01;31:*.t7z=01;31:*.tar=01;31:*.taz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tgz=01;31:*.tlz=01;31:*.txz=01;31:*.tz=01;31:*.tzo=01;31:*.tzst=01;31:*.udeb=01;31:*.war=01;31:*.whl=01;31:*.wim=01;31:*.xz=01;31:*.z=01;31:*.zip=01;31:*.zoo=01;31:*.zst=01;31:*.avif=01;35:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35"..., "MAIL=/var/mail/root", "NIXOS_ACTION=switch", "NIX_PATH=nixpkgs=flake:nixpkgs", "NO_AT_BRIDGE=1", "OUT=/nix/store/xj3icmv6yp869q7x0bvkvm5g3jda9b1r-nixos-system-wsl-24.11.20240612.57d6973", "PATH=/run/wrappers/bin:/home/sedlund/.nix-profile/bin:/nix/profile/bin:/home/sedlund/.local/state/nix/profile/bin:/etc/profiles/per-user/sedlund/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:.:/home/sedlund/bin", "PWD=/nix/store/cvpvxia61l0hxs99lifz0pawa7hpbckm-activatable-nixos-system-wsl-24.11.20240612.57d6973", "QTWEBKIT_PLUGIN_PATH=/root/.nix-profile/lib/mozilla/plugins/:/nix/profile/lib/mozilla/plugins/:/root/.local/state/nix/profile/lib/mozilla/plugins/:/etc/profiles/per-user/root/lib/mozilla/plugins/:/nix/var/nix/profiles/default/lib/mozilla/plugins/:/run/current-system/sw/lib/mozilla/plugins/", "SHELL=/nix/store/3vwx4g1b59z0kzivcmadbjzn3ar1l488-wrapped-bash/wrapper", "SHLVL=0", "SUDO_COMMAND=/run/current-system/sw/bin/strace -f -v -s999 ./bin/switch-to-configuration switch", "SUDO_GID=100", "SUDO_UID=1000", "SUDO_USER=sedlund", "SYSTEMD=/nix/store/xjiifrz7ha6s29gp0p0j3w0155phxmia-systemd-255.6", "TERM=tmux-256color", "TERMINFO_DIRS=/home/sedlund/.nix-profile/share/terminfo:/nix/profile/share/terminfo:/home/sedlund/.local/state/nix/profile/share/terminfo:/etc/profiles/per-user/sedlund/share/terminfo:/nix/var/nix/profiles/default/share/terminfo:/run/current-system/sw/share/terminfo", "TOPLEVEL=/nix/store/xj3icmv6yp869q7x0bvkvm5g3jda9b1r-nixos-system-wsl-24.11.20240612.57d6973", "TZ=Europe/Budapest", "TZDIR=/etc/zoneinfo", "USER=root", "XCURSOR_PATH=/root/.icons:/root/.local/share/icons:/root/.nix-profile/share/icons:/root/.nix-profile/share/pixmaps:/nix/profile/share/icons:/nix/profile/share/pixmaps:/root/.local/state/nix/profile/share/icons:/root/.local/state/nix/profile/share/pixmaps:/etc/profiles/per-user/root/share/icons:/etc/profiles/per-user/root/share/pixmaps:/nix/var/nix/profiles/default/share/icons:/nix/var/nix/profiles/default/share/pixmaps:/run/current-system/sw/share/icons:/run/current-system/sw/share/pixmaps", "XDG_CONFIG_DIRS=/etc/xdg:/root/.nix-profile/etc/xdg:/nix/profile/etc/xdg:/root/.local/state/nix/profile/etc/xdg:/etc/profiles/per-user/root/etc/xdg:/nix/var/nix/profiles/default/etc/xdg:/run/current-system/sw/etc/xdg", "XDG_DATA_DIRS=/root/.nix-profile/share:/nix/profile/share:/root/.local/state/nix/profile/share:/etc/profiles/per-user/root/share:/nix/var/nix/profiles/default/share:/run/current-system/sw/share", "XDG_RUNTIME_DIR=/run/user/1000", "__NIXOS_SWITCH_TO_CONFIGURATION_PARENT_EXE=/nix/store/w5qgrjabiwh5ijm5vmc1kcpn04jfj3nw-switch-to-configuration-0.1.0/bin/switch-to-configuration"]) = -1 EACCES (Permission denied)
</details>

**Comment by jmbaur** [comment]
Did switch-to-configuration-ng work prior to changing usernames under wsl? I'm going to spin up a windows machine so I can try and reproduce.

**Comment by sedlund** [comment]
I don't recall.  The currently released image is still 23.11, and the first thing I did was to nixos-rebuild my flake onto it username and upgrade it to unstable.  without reading how to do the name change.  This was several weeks ago probably before the *ng switcher was added?

I didn't get around to digging into this specifically until today.

everything on the wsl machine seems ok.  I tend to think this is just my problem as I don't see any related issues in the nixos-wsl issue tracker.   

I would try to re-deploy from the 23.11 image and do the upgrade again and properly change the username to test, but I'm currently situated with a very slow internet connection for a bit.
